src/libsystemd-network/network-internal.c

```diff
@@ -40,15 +40,18 @@ const char *net_get_name(sd_device *device) {
 
 int net_get_unique_predictable_data(sd_device *device, uint64_t *result) {
         size_t l, sz = 0;
-        const char *name = NULL;
+        const char *name;
         int r;
         uint8_t *v;
 
         assert(device);
 
         name = net_get_name(device);
         if (!name)
-                return -ENOENT;
+                if (sd_device_get_property_value(device, "ID_BUS", NULL) == -ENOENT)
```

Comment:
I am not sure i understand the rationale for this check. I mean, if a device has zero identifying info, but is a hw device, why not just use the "eth0" name? I mean, giving eth0 (whatever it might be) the same MAC address sounds like an OK thing to me, no?

I mean, I like simplicity: i'd really just use the sysname here if no proper identification is available, and doesn't sound so bad to me, not sure why adding the extra complexity of the if check here...